src/models/application.ts

```diff
@@ -455,40 +451,51 @@ const getApplicationModel = function (
 		 * 	console.log(device);
 		 * })
 		 */
-		async getWithDeviceServiceDetails(
-			slugOrUuidOrId: string | number,
-			options?: PineOptions<Application>,
-		): Promise<
-			Application & {
-				owns__device: Array<DeviceWithServiceDetails<CurrentServiceWithCommit>>;
-			}
-		> {
-			options ??= {};
-			const serviceOptions = mergePineOptions(
-				{
-					$expand: [
-						{
-							owns__device: {
-								$expand: getCurrentServiceDetailsPineExpand(true),
-							},
-						},
-					],
+		async getWithDeviceServiceDetails<
+			T extends ODataOptionsWithoutCount<Application['Read']>,
+		>(slugOrUuidOrId: string | number, options?: T) {
+			const baseOptions = {
+				$expand: {
+					owns__device: {
+						$expand: getCurrentServiceDetailsPineExpand,
+					},
 				},
-				options,
-			);
+			} as const;
+
+			const serviceOptions = mergePineOptions(baseOptions, options);
 
 			const app = (await exports.get(
 				slugOrUuidOrId,
 				serviceOptions,
-			)) as Application & {
-				owns__device: Array<DeviceWithServiceDetails<CurrentServiceWithCommit>>;
-			};
+			)) as unknown as NonNullable<
+				OptionsToResponse<
+					Application['Read'],
+					typeof baseOptions,
+					typeof slugOrUuidOrId
+				>
+			>;
```

Comment:
I thought that if we did the cast upfront, then we could avoid the ts-expect-errors, but I was getting max call stack excheeded errors when I tried it :(
```suggestion
			)) as unknown as NonNullable<
				OptionsToResponse<
					Application['Read'],
					MergePineOptions<Application['Read'], typeof baseOptions, T>,
					typeof slugOrUuidOrId
				>
			> & {
				owns__device: Array<
					(typeof app)['owns__device'][number] & {
						current_services: Record<string, CurrentServiceWithCommit[]>;
					}
				>;
			};
```